### DISCUSSION TITLE
prefixComponent is only applied on detail views

### DISCUSSION THREAD

**Discussion by ianrobertsFF** [laravel/nova-issues#4343] [Q&A] [answered]
- Laravel Version: 9.14.1
- Nova Version: 4.7.0
- PHP Version: 8.1.4
- Database Driver & Version: N/A
- Operating System and Version: N/A
- Browser type and version: N/A

### Description:
The prefixComponent meta is only applied on the Detail View:

```
:is="resolveComponentName(panel)"
```

```
    resolveComponentName(panel) {
      return isNil(panel.prefixComponent) || panel.prefixComponent
        ? 'detail-' + panel.component
        : panel.component
    },
```

On Index, Update, and Create:

```
:is="'form-' + panel.component"
```

It would be more consistent, and more useful, if prefixComponent were applied regardless of the resource view.

### Detailed steps to reproduce the issue on a fresh Nova installation:
1) Install Nova
2) Try to use prefixComponent on anything other than detail

**Comment by crynobone** [comment, marked as answer]
This isn't a bug, `resolveComponentName` is used specifically for Resource Tool which only available on Detail view. 

If you want to suggest that we add it for every view submit an idea via <https://github.com/laravel/nova-issues/discussions/new?category=ideas-feature-requests>

**Reply by ianrobertsFF** [reply]
Fair enough, will do, thanks!
